packages/gatsby/src/query-runner/pages-writer.js

```diff
@@ -2,6 +2,18 @@ const _ = require(`lodash`)
 const glob = require(`glob`)
 const parseFilepath = require(`parse-filepath`)
 const fs = require(`fs`)
+const path = require('path')
+const os = require('os');
+
+function joinPath(...paths) {
+  const joinedPath = path.join(...paths)
+  if(os.platform() === 'win32') {
```

Comment:
Actually I think `path.join` is all we need https://nodejs.org/api/path.html#path_path_join_paths